classes/Project/ProjectFactory.mjs

```diff
@@ -0,0 +1,86 @@
+import {Page} from "../Page/Page.mjs"
+import Project from "./Project.mjs"
+
+let err_out = Object.assign(new Error(), {
+  status: 500,
+  message: "Unknown Server error"
+})
+
+export default class ProjectFactory {
+  constructor(data) {
+    this.data = data
+  }
+
+  static async fetchManifest(url) { 
+    return fetch(url)
+      .then((response) => {
+        return response.json()
+      })
+      .catch((err) => {
+        err_out.status = err.status??404
+        err_out.message = err.message??"Manifest not found. Please check URL" 
+        console.log(err)
+        throw err_out
+      })
+  }
+
+  static async processManifest(manifest) {
+    if (!manifest) {
+      err_out.status = 404
+      err_out.message = "No manifest found. Cannot process empty object"
+      throw err_out
+    }
+    let newProject = {}
+    newProject.title = manifest.label
+    newProject.metadata = manifest.metadata
+
+    newProject["@context"] = "http://t-pen.org/3/context.json"
+    newProject.manifest = manifest["@id"] ?? manifest.id
```

Comment:
Not required now, but this is probably where we would want to start considering using manifesto or vault to access these types of things.

Reply:
Ugh, definitely.  I will need to learn them too.